src/vs/workbench/contrib/notebook/common/services/notebookSimpleWorker.ts

```diff
@@ -56,7 +56,8 @@ class MirrorCell {
 		hashValue = doHash(this.language, hashValue);
 		hashValue = doHash(this.getValue(), hashValue);
 		hashValue = doHash(this.metadata, hashValue);
-		hashValue = doHash({ ...this.internalMetadata, 'cellId': '' }, hashValue);
+		// For purpose of diffing only cellId matters, rest do not
+		hashValue = doHash(this.internalMetadata?.cellId || '', hashValue);
```

Comment:
@amunger If internal metadata, such as cell state or anything else thats internal changes, then this cell is marked as changed.
For the purpose of diffing none of the values matter, in face prior to your PR, internal metadata wasn't synced with worker, hence this is a valid change.
```
	runStartTimeAdjustment?: number;
	runEndTime?: number;
	renderDuration?: { [key: string]: number };
```